File: 2025-1/sesion-de-clase-07/SoftProg/SoftProgRRHHModel/src/main/java/pe/edu/pucp/inf30/softprog/rrhh/model/Persona.java
package pe.edu.pucp.inf30.softprog.rrhh.model;

import java.util.Date;

/**
 *
 * @author eric
 */
public class Persona extends ModeloBase {
    private Area area;
    private CuentaUsuario cuentaUsuario;
    private int dni;
    private String nombre;
    private String apellidoPaterno;
    private char genero;
    private Date fechaNacimiento;

    public Area getArea() {
        return area;
    }

    public void setArea(Area area) {
        this.area = area;
    }

    public CuentaUsuario getCuentaUsuario() {
        return cuentaUsuario;
    }

    public void setCuentaUsuario(CuentaUsuario cuentaUsuario) {
        this.cuentaUsuario = cuentaUsuario;
    }

    public int getDni() {
        return dni;
    }

    public void setDni(int dni) {
        this.dni = dni;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellidoPaterno() {
        return apellidoPaterno;
    }

    public void setApellidoPaterno(String apellidoPaterno) {
        this.apellidoPaterno = apellidoPaterno;
    }

    public char getGenero() {
        return genero;
    }

    public void setGenero(char genero) {
        this.genero = genero;
    }

    public Date getFechaNacimiento() {
        return fechaNacimiento;
    }

    public void setFechaNacimiento(Date fechaNacimiento) {
        this.fechaNacimiento = fechaNacimiento;
    }
}

File: 2026-2/sesion-de-clase-11/SoftProg/SoftProg.Modelo/Persona.cs
using SoftProg.Modelo.Seguridad;

namespace SoftProg.Modelo {
    public class Persona : Registro {
        public Persona() { }
        public Persona(Persona persona) : base(persona) { 

        }

        public CuentaUsuario? CuentaUsuario {
            get {
                return field != null ? new CuentaUsuario(field) : null;
            }
            set {
                field = value != null ? new CuentaUsuario(value) : null;
            }
        }

        public string Dni { 
            get;
            set {
                ArgumentNullException.ThrowIfNullOrEmpty(value);
                field = value;
            }
        }

        public string AppellidoPaterno {
            get;
            set {
                ArgumentNullException.ThrowIfNullOrEmpty(value);
                field = value;

            }
        }
        
        public Genero Genero { get; set; }
        public DateTime FechaNacimiento { get; set; }
    }
}
